### PR TITLE
Introduce re-sort page so claimants can change the category of a prev…

### PR DESCRIPTION
**Pull request by nathancashmoredwp**
…iously sorted activity.

### CODE REVIEW

**Comment by nathancashmoredwp** [comment]
@jessstubenbord can you take a look at this and let me know if your happy with it.

**Comment by lukasz-kaniowski** [comment]
After you re-categorize activity user should probably be redirected back to re-categorize list page, right?

File: app/controllers/categorise-activity-controller.js [hunk 1]
```diff
@@ -13,6 +13,15 @@ function getActivityTitle(activityId) {
   return i18n.__(`activity.${activityId}.title`);
 }
 
+function upsertActivityCategory(currentActivityModel, category) {
```
Comment (lukasz-kaniowski) [inline app/controllers/categorise-activity-controller.js]: Could be moved to `ActivityModel` or some new service?

File: test/integration/categorise-activity.spec.js [hunk 2]
```diff
@@ -53,4 +53,21 @@ describe('Categorise activity page', () => {
...
+  describe('re-categorise activity', () => {
+    before(() =>
+      helper.cleanDb()
+        .then(() => helper.addSortedActivities(accountId, [
+          { activity, category: 'READY' },
+        ]))
+        .then(() => categoriseActivityPage.visit(accountId, activity))
+    );
+
+    it('should maintain the fact an activity has only one category', (done) =>
```
Comment (lukasz-kaniowski) [inline test/integration/categorise-activity.spec.js]: `done` should not be required, looks that everything is a promise so as long as the test method returns a promise it should be tip-top

File: test/integration/re-sort-activities.spec.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+      );
+    });
+
+    it('should link to categorise activity page for activity', () =>
+      reSortActivitiesPage.clickCategoriseButton(allActivites[0]).then(() =>
+        expect(helper.categoriseActivityPage.browserPath())
+          .to.contain(`${accountId}/activities/${allActivites[0].name}/categorise`)
+      )
+    );
+  });
```
Comment (lukasz-kaniowski) [inline test/integration/re-sort-activities.spec.js]: would be good to have a test that exercise the whole re-sort flow: `resort list page -> categories page -> redirect back`

File: app/controllers/categorise-activity-controller.js (hunk 1 above)
Comment (nathancashmoredwp) [inline app/controllers/categorise-activity-controller.js]: ActivityModel makes sense ... will add it there.

File: test/integration/categorise-activity.spec.js (hunk 2 above)
Comment (nathancashmoredwp) [inline test/integration/categorise-activity.spec.js]: Good spot... was a hangover from a restructure of that test.

File: test/integration/re-sort-activities.spec.js (hunk 3 above)
Comment (nathancashmoredwp) [inline test/integration/re-sort-activities.spec.js]: Originally was going to do this in the re-sort spec but fleshing it out showed it was really just an addition test required in the categorise-activity.spec.

Will add an additional test which is required as part of redirecting back from re categorisation after an upsert.  This i think covers it as its `known state -> categories page -> redirect back`

File: app/models/activity-model.js [hunk 4]
```diff
@@ -31,5 +31,15 @@ module.exports = db.Model.extend(
         });
     },
 
+    updateCategorisation(accountId, activity, category) {
+      return this.forge()
+        .query({ where: { accountId, activity } }).fetch()
+        .then((modelFound) => {
+          if (modelFound) {
+            return modelFound.save({ category }).then(() => 'UPDATE');
```
Comment (lukasz-kaniowski) [inline app/models/activity-model.js]: looks that this method returns inconsistent results. what about something like:
`{ status: 'UPDATE', model: updatedModel} `
and 
`{ status: 'NEW', model: newModel} `
if you need model at all.

Comment (nathancashmoredwp) [inline app/models/activity-model.js]: I did have that in there originally ... but without the model.  Check later is just on the UPDATE but agree looking at it again returning something that is more explicit would be clearer... will change to return ``{ status: 'UPDATE' }`` and ``{ status: 'NEW' }``

**Comment by lukasz-kaniowski** [comment]
👍

**Pull request merged**